src/code/z_parameter.c

```diff
@@ -722,15 +722,16 @@ void func_80083108(PlayState* play) {
                 Interface_ChangeAlpha(50);
             }
         } else if (msgCtx->msgMode == MSGMODE_NONE) {
-            if ((func_8008F2F8(play) >= 2) && (func_8008F2F8(play) < 5)) {
+            if ((Player_GetEnvTimerType(play) >= PLAYER_ENV_TIMER_TYPE_UNDERWATER_FLOOR) &&
+                (Player_GetEnvTimerType(play) <= PLAYER_ENV_TIMER_TYPE_UNDERWATER_FREE)) {
                 if (gSaveContext.buttonStatus[0] != BTN_DISABLED) {
                     sp28 = true;
                 }
 
                 gSaveContext.buttonStatus[0] = BTN_DISABLED;
 
                 for (i = 1; i < 4; i++) {
-                    if (func_8008F2F8(play) == 2) {
+                    if (Player_GetEnvTimerType(play) == PLAYER_ENV_TIMER_TYPE_UNDERWATER_FLOOR) {
                         if ((gSaveContext.equips.buttonItems[i] != ITEM_HOOKSHOT) &&
                             (gSaveContext.equips.buttonItems[i] != ITEM_LONGSHOT)) {
                             if (gSaveContext.buttonStatus[i] == BTN_ENABLED) {
```

Comment:
:spaghetti: 
There are a few spots that are totally unrelated to timers, like here for enabling hookshot underwater

Maybe there's a more generic term than "env timer type" we could find? something with "hazard" maybe?

Reply:
I actually really like the `hazard` term, so `Player_GetEnvHazardType`? I think of enemies and other obstacles as hazards too, but I think it would still need `env` or an alternative? Maybe even drop the `Type`

Reply:
Sure, EnvHazard sounds fine

it's not like this covers every hazard anyway, it's an adventure game after all, but at least the term is new and I don't think it would be better reserved for something else